task-ballerina/commons.bal

```diff
@@ -14,118 +14,61 @@
+public type Job object {
 
-# Possible types of parameters that can be passed into the `AppointmentTaskPolicy`.
-public type AppointmentMisfirePolicy SMART_POLICY|DO_NOTHING|FIRE_AND_PROCEED|IGNORE_POLICY;
+  # Executes by the Scheduler when the scheduled trigger fires.
+  public function execute();
+};
 
-# Possible types of parameters that can be passed into the `RecurringTaskPolicy`.
-public type RecurringTaskPolicy SMART_POLICY|IGNORE_POLICY|FIRE_NEXT_WITH_EXISTING_COUNT|
-FIRE_NEXT_WITH_REMAINING_COUNT|FIRE_NOW_WITH_EXISTING_COUNT|FIRE_NOW_WITH_REMAINING_COUNT;
+# Policies related to a trigger.
+#
+# + errorPolicy - The policy to follow when there is an error in Job execution
+# + waitingPolicy - The policy to follow when the next task is triggering while the previous job is still
+#                   being processing
+public type TaskPolicy record {|
+   ErrorPolicy errorPolicy = LOG_AND_TERMINATE;
+   WaitingPolicy waitingPolicy = WAIT;
+|};
+
+# Possible options for the `ErrorPolicy`.
+public enum ErrorPolicy {
+  LOG_AND_TERMINATE,
+  LOG_AND_CONTINUE,
+  TERMINATE,
+  CONTINUE
+}
 
-# Possible types of parameters that can be passed into the `OneTimeTaskPolicy`.
-public type OneTimeTaskPolicy SMART_POLICY|IGNORE_POLICY|FIRE_NOW;
+# Possible options for the `WaitingPolicy`.
+public enum WaitingPolicy {
+  WAIT,
+  IGNORE,
+  LOG_AND_IGNORE
+}
 
-# The value of this constant is `smartPolicy`.
-public const string SMART_POLICY = "smartPolicy";
-# The value of this constant is `doNothing`.
-public const string DO_NOTHING = "doNothing";
-# The value of this constant is `fireAndProceed`.
-public const string FIRE_AND_PROCEED = "fireAndProceed";
-# The value of this constant is `fireNextWithExistingCount`.
-public const string FIRE_NEXT_WITH_EXISTING_COUNT = "fireNextWithExistingCount";
-# The value of this constant is `fireNextWithRemainingCount`.
-public const string FIRE_NEXT_WITH_REMAINING_COUNT = "fireNextWithRemainingCount";
-# The value of this constant is `fireNowWithExistingCount`.
-public const string FIRE_NOW_WITH_EXISTING_COUNT = "fireNowWithExistingCount";
-# The value of this constant is `fireNowWithRemainingCount`.
-public const string FIRE_NOW_WITH_REMAINING_COUNT = "fireNowWithRemainingCount";
-# The value of this constant is `fireNow`.
-public const string FIRE_NOW = "fireNow";
-# The value of this constant is `ignorePolicy`.
-public const string IGNORE_POLICY = "ignorePolicy";
+# Gets time in milliseconds of the given `time:Civil`.
+#
+# + time - The Ballerina `time:Civil`
+# + return - Time in milliseconds or else `task:Error` if the process failed due to any reason
+public isolated function getTimeInMillies(time:Civil time) returns int|Error {
+    if (time[UTC_OFF_SET] == ()) {
+        time:ZoneOffset zoneOffset = {hours: 0, minutes: 0};
+        time.utcOffset = zoneOffset;
+    }
+    time:Utc|time:Error utc = time:utcFromCivil(time);
+    if (utc is time:Utc) {
+        return <int> ((utc[0] + utc[1]) * 1000);
```

Comment:
Converting this to int will lose some of the information. It is okay to lose these nanoseconds level information since we are converting to milliseconds. However, instead of typecasting, I would suggest using the decimal round(https://ballerina.io/learn/api-docs/ballerina/#/ballerina/lang.decimal/1.0.0/lang.decimal/functions#round).